src/storage.rs

```diff
@@ -1,365 +1,92 @@
-    // Files
-    let files_table_cmd = format!(
-        "CREATE TABLE IF NOT EXISTS {} (
-        id TEXT PRIMARY KEY,
-        timestamp INTEGER
-    )",
-        FILES_TABLE
-    );
-    conn.execute(&files_table_cmd, params![]).expect("Couldn't create files table.");
-    // User activity table
-    let user_activity_table_cmd = format!(
-        "CREATE TABLE IF NOT EXISTS {} (
-        public_key TEXT PRIMARY KEY,
-        last_active INTEGER NOT NULL
-    )",
-        USER_ACTIVITY_TABLE,
-    );
-    conn.execute(&user_activity_table_cmd, params![])
-        .expect("Couldn't create user activity table.");
+    // Future DB migration code goes here
 }
 
 // Pruning
 
-pub async fn prune_tokens_periodically() {
-    let mut timer = tokio::time::interval(chrono::Duration::minutes(10).to_std().unwrap());
-    loop {
-        timer.tick().await;
-        tokio::spawn(async {
-            prune_tokens().await;
-        });
-    }
-}
-
-pub async fn prune_pending_tokens_periodically() {
-    let mut timer = tokio::time::interval(chrono::Duration::minutes(10).to_std().unwrap());
-    loop {
-        timer.tick().await;
-        tokio::spawn(async {
-            prune_pending_tokens().await;
-        });
-    }
-}
-
 pub async fn prune_files_periodically() {
-    let mut timer = tokio::time::interval(chrono::Duration::days(1).to_std().unwrap());
+    let mut timer = tokio::time::interval(chrono::Duration::seconds(15).to_std().unwrap());
     loop {
         timer.tick().await;
         tokio::spawn(async {
-            prune_files(FILE_EXPIRATION).await;
+            prune_files(SystemTime::now()).await;
         });
     }
 }
 
-async fn prune_tokens() {
-    let rooms = match get_all_room_ids() {
-        Ok(rooms) => rooms,
-        Err(_) => return,
-    };
-    for room in rooms {
-        let pool = pool_by_room_id(&room);
-        // It's not catastrophic if we fail to prune the database for a given room
-        let conn = match pool.get() {
-            Ok(conn) => conn,
-            Err(e) => return error!("Couldn't prune tokens due to error: {}.", e),
-        };
-        let stmt = format!("DELETE FROM {} WHERE timestamp < (?1)", TOKENS_TABLE);
-        let now = chrono::Utc::now().timestamp();
-        let expiration = now - TOKEN_EXPIRATION;
-        match conn.execute(&stmt, params![expiration]) {
-            Ok(_) => (),
-            Err(e) => return error!("Couldn't prune tokens due to error: {}.", e),
-        };
-    }
-    info!("Pruned tokens.");
-}
+/// Removes all files with expiries <= the given time (which should generally by
+/// `SystemTime::now()`, except in the test suite).
+pub async fn prune_files(now: SystemTime) {
 
-async fn prune_pending_tokens() {
-    let rooms = match get_all_room_ids() {
-        Ok(rooms) => rooms,
-        Err(_) => return,
+    let conn = match DB_POOL.get() {
+        Ok(conn) => conn,
+        Err(e) => return error!("Couldn't prune files: {}.", e)
```

Comment:
I realise that this wasn't introduced in this PR but worth improving this since it is being touched:
```suggestion
        Err(e) => {
            error!("Couldn't prune files: {}.", e);
            return;
        }
```
Otherwise it looks like we are "returning" the error which isn't the case...

Reply:
I grabbed that from other code but yeah, I agree.